File: src/fastqToFasta.pl
#!/usr/bin/env perl
#
#convert fastq on STDIN  to fasta on STDOUT, leave untouched if fasta

while($line=<STDIN>){
  if(substr($line,0,1) eq "@"){ #type is fastq
    print ">",substr($line,1); 
    $seq=""; 
    $nlines=0;
    while($line=<STDIN>){
      chomp($line); 
      last if(substr($line,0,1) eq "+");
      $seq.=$line;
      $nlines++;
    } 
    print $seq,"\n"; 
    $slines=0;
    while($line=<STDIN>){
      $slines++; 
      last if($slines==$nlines);
    }
  }elsif(substr($line,0,1) eq ">"){ #type is fasta, output unchanged
    print $line;
    while($line=<STDIN>){
      print $line;
    }
  }
}

File: src/fastqToFasta.py
import sys

def convert_fastq_to_fasta():
    for line in sys.stdin:
        if line.startswith("@"):  # type is fastq
            print(">" + line[1:].strip()) 
            seq = ""
            nlines = 0
            for line in sys.stdin:
                line = line.strip()
                if line.startswith("+"):
                    break
                seq += line
                nlines += 1
            print(seq)
            slines = 0
            for line in sys.stdin:
                slines += 1
                if slines == nlines:
                    break
        elif line.startswith(">"):  # type is fasta, output unchanged
            print(line.strip())
            for line in sys.stdin:
                print(line.strip())

if __name__ == "__main__":
    convert_fastq_to_fasta()
